src/Workspaces/Remote/ServiceHub/Shared/ServiceHubServiceBase.cs

```diff
@@ -46,6 +47,26 @@ protected ServiceHubServiceBase(Stream stream, IServiceProvider serviceProvider)
             Rpc.Disconnected += OnRpcDisconnected;
         }
 
+        protected ServiceHubServiceBase(IServiceProvider serviceProvider, Stream stream)
```

Comment:
why have this and the above constructor?  Why just change the above constructor to call "new JsonRpc" instead of "JsonRpc.Attach"?